egui/src/widgets/plot/mod.rs

```diff
@@ -122,6 +122,19 @@ impl Plot {
         self
     }
 
+    pub fn lines(mut self, lines: Vec<Line>) -> Self {
+        for mut line in lines{
+            if line.series.is_empty(){
+                return self;
+            }
+            if line.stroke.color == Color32::TRANSPARENT {
+                line.stroke.color = self.auto_color();
+            }
+            self.items.push(Box::new(line));
+        }
```

Comment:
No need for code duplication

```suggestion
        for line in lines {
            self = self.line(line);
        }
```

though I would argue the caller could just do this instead?